### PR TITLE
Deploy to heroku

### PR DESCRIPTION
**Pull request by felipesere**

### CODE REVIEW

File: config/dev.exs [hunk 1]
```diff
@@ -55,3 +55,8 @@ config :advisor, Advisor.Repo,
   database: "advisor_dev",
   hostname: "localhost",
   pool_size: 10
+
+config :advisor, Advisor.Web.Authentication.Password,
```
Comment (fabientownsend) [inline config/dev.exs]: Shouldn't it belong to Core rather than Web?

File: config/prod.exs [hunk 2]
```diff
@@ -15,50 +15,22 @@ use Mix.Config
...
 config :advisor, Advisor.Web.Endpoint,
   on_init: {Advisor.Web.Endpoint, :load_from_system_env, []},
-  url: [host: "example.com", port: 80],
-  cache_static_manifest: "priv/static/cache_manifest.json"
+  url: [scheme: "https", host: "advisorex.herokuapp.com", port: 443],
+  force_ssl: [rewrite_on: [:x_forwarded_proto]],
+  cache_static_manifest: "priv/static/cache_manifest.json",
+  secret_key_base: System.get_env("SECRET_KEY_BASE")
 
 # Do not print debug messages in production
```
Comment (fabientownsend) [inline config/prod.exs]: It's not part of the PR but it make me think, it must have a way to configure that to only print in dev env'?

File: lib/advisor/web/authentication/password.ex [hunk 3]
```diff
@@ -0,0 +1,20 @@
+defmodule Advisor.Web.Authentication.Password do
+  alias Comeonin.Bcrypt
+  def matches?(password) do
+    [checker: checker, password: real_hash] = Application.get_env(:advisor, __MODULE__)
+
+    checker.matches?(password, real_hash)
```
Comment (fabientownsend) [inline lib/advisor/web/authentication/password.ex]: What do you think to call `real_hash` -> `hashed_password`
I mean, I am not sure `real` give really any information as there aren't any `fake_hash` :D

File: lib/advisor/web/controllers/login_controller.ex [hunk 4]
```diff
@@ -1,25 +1,26 @@
 defmodule Advisor.Web.LoginController do
   use Advisor.Web, :controller
   alias Advisor.Core.People
+  alias Advisor.Web.Authentication.Password
 
-  def index(conn, %{"email" => email, "submit" => "dashboard"}) do
-    login(conn, as: email, redirect_to: "/dashboard")
+  def index(conn, %{"email" => email, "password" => pw, "submit" => "dashboard"}) do
```
Comment (fabientownsend) [inline lib/advisor/web/controllers/login_controller.ex]: Wouldn't start with `"submit" =>` make it easier to catch the difference between the methods?

File: config/dev.exs (hunk 1 above)
Comment (felipesere) [inline config/dev.exs]: I wanted to keep it close to the rest of the authentication stuff.

File: config/prod.exs (hunk 2 above)
Comment (felipesere) [inline config/prod.exs]: Yes, that is the way it is configured at the moment. 
There are 3 config files: `prod.exs`, `dev.exs`, and `test.exs`. I believe dev prints, but test doesn't.

Comment (fabientownsend) [inline config/prod.exs]: Oh right, so the comment was more saying what is the current behaviour! I read it like, don't push any print in prod!

File: config/dev.exs (hunk 1 above)
Comment (felipesere) [inline config/dev.exs]: I'm also adding a README in that space too 😸
